android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -166,10 +193,89 @@ public object WorkerBinder {
   }
 }
 
+/**
+ * Threading type where Worker.onStart/onStop will be running upon
+ */
+public enum class WorkerBinderThreadingType {
+  BACKGROUND_COMPUTATION,
+  BACKGROUND_IO,
+  MAIN,
+  CALLER_THREAD
+}
+
+/**
+ * Worker.onStart/onStop duration information
+ */
+public data class WorkerBinderDuration(
+  /**
+   * Worker class name
+   */
+  val workerName: String,
+
+  /**
+   * Worker event type (START/STOP)
+   */
+  val workerEvent: WorkerEvent,
+
+  /**
+   * Threading type where Worker.onStart/onStop upon calling [WorkerBinder.bind]
+   */
+  val workerBinderThreadingType: WorkerBinderThreadingType,
```

Comment:
Since the data doesn't actually tell us which thread, just if it was calling thread of parent, do we want to track that here, or will be in a custom impl internally?

Reply:
Was thinking to use `threadName` within the `WorkerBinderDuration` data class to be called on each onStart/onStop completion